packages/sw-lib/src/lib/error-factory.js

```diff
@@ -0,0 +1,61 @@
+/*
+ Copyright 2016 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+const errors = {
+  'not-in-sw': 'sw-lib must be loaded in your service worker file.',
+  'unsupported-route-type': 'Routes must be either a express style route ' +
+    'string, a Regex to capture request URLs or a Route instance.',
+  'empty-express-string': 'The Express style route string must have some ' +
+    'characters, an empty string is invalid.',
+};
+
+/**
+ * A simple class to make errors and to help with testing.
+ */
+class ErrorFactory {
+  /**
+   * @param {string} name The error name to be generated.
+   * @param {Error} [thrownError] The thrown error that resulted in this
+   * message.
+   * @return {Error} The generated error.
+   */
+  static createError(name, thrownError) {
```

Comment:
Elsewhere, we're using modules with named exports, which accomplishes the same thing as a Factory class with entirely static methods: https://github.com/GoogleChrome/sw-helpers/blob/master/lib/assert.js

I'm in favor of the module + named exports approach, as selectively importing what you need facilitates tree-shaking, and I think it feels more like "modern" JavaScript (for whatever that's worth).